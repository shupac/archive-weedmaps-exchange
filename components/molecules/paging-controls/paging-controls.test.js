import React from 'react';
import { shallow, mount } from 'enzyme';

import PagingControls from './';

const pageButtonArray = (pageCount, currentPage) => {
  const wrapper = shallow(
    <PagingControls pageCount={pageCount} currentPage={currentPage} />,
  );
  const componentInstance = wrapper.instance();
  return componentInstance.pageButtonArray();
};
describe('Paging Controls', () => {
  it('should render with four pages', () => {
    const component = <PagingControls pageCount={4} />;
    const tree = shallow(component);
    expect(tree.exists()).toEqual(true);
  });

  it('should render with nine pages', () => {
    const component = <PagingControls pageCount={9} currentPage={5} />;
    const tree = shallow(component);
    expect(tree.exists()).toEqual(true);
  });

  it('should render with 0 pages', () => {
    const component = <PagingControls pageCount={0} />;
    const tree = shallow(component);
    expect(tree.exists()).toEqual(true);
  });

  it('should have the right page number values with four pages', () => {
    expect(pageButtonArray(4, 1)).toEqual([1, 2, 3, 4]);
    expect(pageButtonArray(4, 2)).toEqual([1, 2, 3, 4]);
    expect(pageButtonArray(4, 3)).toEqual([1, 2, 3, 4]);
    expect(pageButtonArray(4, 4)).toEqual([1, 2, 3, 4]);
  });

  it('should have the right page number values with seven pages', () => {
    expect(pageButtonArray(7, 1)).toEqual([1, 2, 3, 4, 5, 6, 7]);
    expect(pageButtonArray(7, 2)).toEqual([1, 2, 3, 4, 5, 6, 7]);
    expect(pageButtonArray(7, 3)).toEqual([1, 2, 3, 4, 5, 6, 7]);
    expect(pageButtonArray(7, 4)).toEqual([1, 2, 3, 4, 5, 6, 7]);
    expect(pageButtonArray(7, 5)).toEqual([1, 2, 3, 4, 5, 6, 7]);
    expect(pageButtonArray(7, 6)).toEqual([1, 2, 3, 4, 5, 6, 7]);
    expect(pageButtonArray(7, 7)).toEqual([1, 2, 3, 4, 5, 6, 7]);
  });

  it('should have the right page number values with nine pages', () => {
    expect(pageButtonArray(9, 1)).toEqual([1, 2, 3, 4, 5, 0, 9]);
    expect(pageButtonArray(9, 2)).toEqual([1, 2, 3, 4, 5, 0, 9]);
    expect(pageButtonArray(9, 3)).toEqual([1, 2, 3, 4, 5, 0, 9]);
    expect(pageButtonArray(9, 4)).toEqual([1, 2, 3, 4, 5, 0, 9]);
    expect(pageButtonArray(9, 5)).toEqual([1, 0, 4, 5, 6, 0, 9]);
    expect(pageButtonArray(9, 6)).toEqual([1, 0, 5, 6, 7, 8, 9]);
    expect(pageButtonArray(9, 7)).toEqual([1, 0, 5, 6, 7, 8, 9]);
    expect(pageButtonArray(9, 8)).toEqual([1, 0, 5, 6, 7, 8, 9]);
    expect(pageButtonArray(9, 9)).toEqual([1, 0, 5, 6, 7, 8, 9]);
  });

  it('should have the right page number values with twelve pages', () => {
    expect(pageButtonArray(12, 1)).toEqual([1, 2, 3, 4, 5, 0, 12]);
    expect(pageButtonArray(12, 2)).toEqual([1, 2, 3, 4, 5, 0, 12]);
    expect(pageButtonArray(12, 3)).toEqual([1, 2, 3, 4, 5, 0, 12]);
    expect(pageButtonArray(12, 4)).toEqual([1, 2, 3, 4, 5, 0, 12]);
    expect(pageButtonArray(12, 5)).toEqual([1, 0, 4, 5, 6, 0, 12]);
    expect(pageButtonArray(12, 6)).toEqual([1, 0, 5, 6, 7, 0, 12]);
    expect(pageButtonArray(12, 7)).toEqual([1, 0, 6, 7, 8, 0, 12]);
    expect(pageButtonArray(12, 8)).toEqual([1, 0, 7, 8, 9, 0, 12]);
    expect(pageButtonArray(12, 9)).toEqual([1, 0, 8, 9, 10, 11, 12]);
    expect(pageButtonArray(12, 10)).toEqual([1, 0, 8, 9, 10, 11, 12]);
    expect(pageButtonArray(12, 11)).toEqual([1, 0, 8, 9, 10, 11, 12]);
    expect(pageButtonArray(12, 12)).toEqual([1, 0, 8, 9, 10, 11, 12]);
  });

  it('should call onSelectPage when clicking on a page number', () => {
    const onSelectPage = jest.fn();
    const wrapper = shallow(
      <PagingControls
        pageCount={4}
        currentPage={3}
        onSelectPage={onSelectPage}
      />,
    );
    const pageButton = wrapper.find('PageButton').first();

    pageButton.simulate('click');
    expect(onSelectPage).toHaveBeenCalledWith(1);
  });

  it('should call onSelectPage when clicking previous page button', () => {
    const onSelectPage = jest.fn();
    const wrapper = mount(
      <PagingControls
        pageCount={4}
        currentPage={4}
        onSelectPage={onSelectPage}
      />,
    );
    wrapper.find('PreviousPageButton').simulate('click');
    expect(onSelectPage).toHaveBeenCalledWith(3);
  });

  it('should call onSelectPage when clicking next page button', () => {
    const onSelectPage = jest.fn();
    const wrapper = mount(
      <PagingControls
        pageCount={4}
        currentPage={1}
        onSelectPage={onSelectPage}
      />,
    );
    wrapper.find('NextPageButton').simulate('click');
    expect(onSelectPage).toHaveBeenCalledWith(2);
  });

  it('should not call onSelectPage when clicking previous page button while on the first page', () => {
    const onSelectPage = jest.fn();
    const wrapper = mount(
      <PagingControls
        pageCount={4}
        currentPage={1}
        onSelectPage={onSelectPage}
      />,
    );
    wrapper.find('PreviousPageButton').simulate('click');
    expect(onSelectPage).not.toHaveBeenCalled();
  });

  it('should not call onSelectPage when clicking next page button while on the last page', () => {
    const onSelectPage = jest.fn();
    const wrapper = mount(
      <PagingControls
        pageCount={4}
        currentPage={4}
        onSelectPage={onSelectPage}
      />,
    );
    wrapper.find('NextPageButton').simulate('click');
    expect(onSelectPage).not.toHaveBeenCalled();
  });
});
