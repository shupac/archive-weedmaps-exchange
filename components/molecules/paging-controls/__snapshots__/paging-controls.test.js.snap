// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render with four pages 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c4 {
  min-width: 2rem;
  height: 2rem;
  font-size: 0.875rem;
  border-radius: 0.1875rem;
  line-height: 1.8125rem;
  border: 0.0625rem solid;
  margin: 0 0.125rem 0 0.125rem;
  cursor: pointer;
}

.c5 {
  cursor: default;
}

.c3 {
  width: 3.75rem;
  padding: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c2 {
  margin: 0 0.375rem 0 0;
}

.c6 {
  margin: 0 0 0 0.375rem;
}

.c1 {
  box-sizing: border-box;
}

<div
  className="c0 c1"
>
  <button
    className="c2 c3 c4"
    onClick={[Function]}
  >
    <svg
      height="14px"
      width="8px"
    >
      <path
        d="M7.533 6.538L8 7l-.467.462L.933 14 0 13.075l6.6-6.537v.924L0 .925.933 0z"
        fill="#999999"
        transform="rotate(180 4 7)"
      />
    </svg>
  </button>
  <div>
    <button
      className="c5 c4"
    >
      1
    </button>
    <button
      className="c4"
      onClick={[Function]}
    >
      2
    </button>
    <button
      className="c4"
      onClick={[Function]}
    >
      3
    </button>
    <button
      className="c4"
      onClick={[Function]}
    >
      4
    </button>
  </div>
  <button
    className="c6 c3 c4"
    onClick={[Function]}
  >
    <svg
      height="14px"
      width="8px"
    >
      <path
        d="M7.533 6.538L8 7l-.467.462L.933 14 0 13.075l6.6-6.537v.924L0 .925.933 0z"
        fill="#999999"
        transform="rotate(0 0 0)"
      />
    </svg>
  </button>
</div>
`;

exports[`should render with nine pages 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c4 {
  min-width: 2rem;
  height: 2rem;
  font-size: 0.875rem;
  border-radius: 0.1875rem;
  line-height: 1.8125rem;
  border: 0.0625rem solid;
  margin: 0 0.125rem 0 0.125rem;
  cursor: pointer;
}

.c5 {
  border: none;
  background-color: transparent;
  cursor: default;
}

.c6 {
  cursor: default;
}

.c3 {
  width: 3.75rem;
  padding: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c2 {
  margin: 0 0.375rem 0 0;
}

.c7 {
  margin: 0 0 0 0.375rem;
}

.c1 {
  box-sizing: border-box;
}

<div
  className="c0 c1"
>
  <button
    className="c2 c3 c4"
    onClick={[Function]}
  >
    <svg
      height="14px"
      width="8px"
    >
      <path
        d="M7.533 6.538L8 7l-.467.462L.933 14 0 13.075l6.6-6.537v.924L0 .925.933 0z"
        fill="#999999"
        transform="rotate(180 4 7)"
      />
    </svg>
  </button>
  <div>
    <button
      className="c4"
      onClick={[Function]}
    >
      1
    </button>
    <button
      className="c5 c4"
    >
      …
    </button>
    <button
      className="c4"
      onClick={[Function]}
    >
      4
    </button>
    <button
      className="c6 c4"
    >
      5
    </button>
    <button
      className="c4"
      onClick={[Function]}
    >
      6
    </button>
    <button
      className="c5 c4"
    >
      …
    </button>
    <button
      className="c4"
      onClick={[Function]}
    >
      9
    </button>
  </div>
  <button
    className="c7 c3 c4"
    onClick={[Function]}
  >
    <svg
      height="14px"
      width="8px"
    >
      <path
        d="M7.533 6.538L8 7l-.467.462L.933 14 0 13.075l6.6-6.537v.924L0 .925.933 0z"
        fill="#999999"
        transform="rotate(0 0 0)"
      />
    </svg>
  </button>
</div>
`;
